sdk/helper/tokenutil/tokenutil.go

```diff
@@ -266,6 +267,133 @@ func DeprecationText(param string) string {
 	return fmt.Sprintf("Use %q instead. If this and %q are both specified, only %q will be used.", param, param, param)
 }
 
+func upgradeDurationValue(d *framework.FieldData, oldKey, newKey string, oldVal, newVal *time.Duration) error {
+	_, ok := d.GetOk(newKey)
+	if !ok {
+		raw, ok := d.GetOk(oldKey)
+		if ok {
+			*oldVal = time.Duration(raw.(int)) * time.Second
+			*newVal = *oldVal
+		}
+	} else {
+		_, ok = d.GetOk(oldKey)
+		if ok {
+			*oldVal = *newVal
+		} else {
+			*oldVal = 0
+		}
+	}
+
+	return nil
+}
+
+func upgradeIntValue(d *framework.FieldData, oldKey, newKey string, oldVal, newVal *int) error {
+	_, ok := d.GetOk(newKey)
+	if !ok {
+		raw, ok := d.GetOk(oldKey)
+		if ok {
+			*oldVal = raw.(int)
+			*newVal = *oldVal
+		}
+	} else {
+		_, ok = d.GetOk(oldKey)
+		if ok {
+			*oldVal = *newVal
+		} else {
+			*oldVal = 0
+		}
+	}
+
+	return nil
+}
+
+func upgradeStringSliceValue(d *framework.FieldData, oldKey, newKey string, oldVal, newVal *[]string) error {
+	_, ok := d.GetOk(newKey)
+	if !ok {
+		raw, ok := d.GetOk(oldKey)
+		if ok {
+			// Special case: if we're looking at "token_policies" parse the policies
+			if newKey == "token_policies" {
```

Comment:
I think this should be `policies` since we parsed the policies in the past for `policies` and not `token_policies`.

Reply:
Old key is "policies", new key is "token_policies". We can' be sure what the old key was, we know for sure what the new key is.

Reply:
Gotcha!